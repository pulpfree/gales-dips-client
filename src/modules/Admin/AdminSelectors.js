import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { withRouter } from 'react-router'

// import Button from '@material-ui/core/Button'
import FormControl from '@material-ui/core/FormControl'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'
import Select from '@material-ui/core/Select'
import { withStyles } from '@material-ui/core/styles'


class AdminSelectors extends Component {
  state = {
    stationID: '',
  }

  componentDidMount = () => {
    const { pathname } = this.props.location
    const stationID = pathname.split('/')[2]

    if (stationID) {
      this.setState({ stationID })
    }
  }

  handleStationChange = (event) => {
    this.setState({ [event.target.name]: event.target.value }, this.handleSetStation)
  }

  handleSetStation = () => {
    const { history, match } = this.props
    const { stationID } = this.state

    if (stationID) {
      const uri = `${match.url}/${stationID}`
      history.push(uri)
    }
  }

  render() {
    const { classes } = this.props

    return (
      <div>
        <FormControl className={classes.formControl}>
          <InputLabel htmlFor="station-select">Station</InputLabel>
          <Select
            inputProps={{
                name: 'stationID',
                id: 'station-select',
              }}
            onChange={this.handleStationChange}
            value={this.state.stationID}
          >
            {mockStations.map(s => (
              <MenuItem
                key={s.id}
                value={s.id}
              >
                {s.name}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </div>
    )
  }
}

AdminSelectors.propTypes = {
  classes: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
}

const styles = theme => ({
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
  },
})

const mockStations = [
  { id: 'b-10', name: 'Bridge' },
  { id: 'ch-20', name: 'Chippawa' },
  { id: 'co-30', name: 'Collier' },
  { id: 'd-40', name: 'Drummond' },
]

export default withRouter(withStyles(styles)(AdminSelectors))
