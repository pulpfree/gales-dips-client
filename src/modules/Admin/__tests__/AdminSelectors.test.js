/* eslint no-undef: "off" */

import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from 'enzyme'

import AdminSelectors from '../AdminSelectors'

describe('AdminSelectors', () => {

  let mountedC
  let props
  const reportC = () => {
    if (!mountedC) {
      mountedC = render(
        <Router><AdminSelectors {...props} /></Router>
      )
    }
    return mountedC
  }

  beforeEach(() => {
    props = {
      // authState: 'confirmSignIn',
    }
    mountedC = undefined
  })

  it('always renders a div', () => {
    const divs = reportC().find('div')
    expect(divs.length).toBeGreaterThan(0)
  })

  it('matches snapshot', () => {
    expect(reportC()).toMatchSnapshot()
  })

})