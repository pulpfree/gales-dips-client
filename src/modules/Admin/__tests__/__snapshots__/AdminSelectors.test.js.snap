// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminSelectors matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiFormControl-root-2 AdminSelectors-formControl-1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiFormLabel-root-13 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10",
              "data-shrink": "false",
              "for": "station-select",
            },
            "children": Array [
              Object {
                "data": "Station",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "MuiInputBase-root-40 MuiInput-root-27 MuiInput-underline-31 MuiInputBase-formControl-41 MuiInput-formControl-28",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "MuiSelect-root-20",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "aria-haspopup": "true",
                        "aria-pressed": "false",
                        "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                        "role": "button",
                        "tabindex": "0",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "â€‹",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "id": "station-select",
                          "name": "stationID",
                          "type": "hidden",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "aria-hidden": "true",
                            "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                            "focusable": "false",
                            "role": "presentation",
                            "viewBox": "0 0 24 24",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "d": "M7 10l5 5 5-5z",
                              },
                              "children": Array [],
                              "name": "path",
                              "namespace": "http://www.w3.org/2000/svg",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "d": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "d": undefined,
                              },
                            },
                          ],
                          "name": "svg",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "aria-hidden": undefined,
                            "class": undefined,
                            "focusable": undefined,
                            "role": undefined,
                            "viewBox": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "aria-hidden": undefined,
                            "class": undefined,
                            "focusable": undefined,
                            "role": undefined,
                            "viewBox": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "id": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "id": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-haspopup": undefined,
                        "aria-pressed": undefined,
                        "class": undefined,
                        "role": undefined,
                        "tabindex": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-haspopup": undefined,
                        "aria-pressed": undefined,
                        "class": undefined,
                        "role": undefined,
                        "tabindex": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "id": "station-select",
                        "name": "stationID",
                        "type": "hidden",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "aria-hidden": "true",
                          "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                          "focusable": "false",
                          "role": "presentation",
                          "viewBox": "0 0 24 24",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "d": "M7 10l5 5 5-5z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "role": undefined,
                          "viewBox": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "role": undefined,
                          "viewBox": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "aria-haspopup": "true",
                          "aria-pressed": "false",
                          "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                          "role": "button",
                          "tabindex": "0",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "â€‹",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-haspopup": undefined,
                          "aria-pressed": undefined,
                          "class": undefined,
                          "role": undefined,
                          "tabindex": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-haspopup": undefined,
                          "aria-pressed": undefined,
                          "class": undefined,
                          "role": undefined,
                          "tabindex": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                        "focusable": "false",
                        "role": "presentation",
                        "viewBox": "0 0 24 24",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "d": "M7 10l5 5 5-5z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "id": "station-select",
                          "name": "stationID",
                          "type": "hidden",
                          "value": "",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "aria-haspopup": "true",
                            "aria-pressed": "false",
                            "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                            "role": "button",
                            "tabindex": "0",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {},
                              "children": Array [
                                Object {
                                  "data": "â€‹",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "span",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {},
                              "x-attribsPrefix": Object {},
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "aria-haspopup": undefined,
                            "aria-pressed": undefined,
                            "class": undefined,
                            "role": undefined,
                            "tabindex": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "aria-haspopup": undefined,
                            "aria-pressed": undefined,
                            "class": undefined,
                            "role": undefined,
                            "tabindex": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "id": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "id": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "role": undefined,
                        "viewBox": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "role": undefined,
                        "viewBox": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-shrink": undefined,
              "for": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-shrink": undefined,
              "for": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "MuiInputBase-root-40 MuiInput-root-27 MuiInput-underline-31 MuiInputBase-formControl-41 MuiInput-formControl-28",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "MuiSelect-root-20",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "aria-haspopup": "true",
                      "aria-pressed": "false",
                      "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                      "role": "button",
                      "tabindex": "0",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {},
                        "children": Array [
                          Object {
                            "data": "â€‹",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "id": "station-select",
                        "name": "stationID",
                        "type": "hidden",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "aria-hidden": "true",
                          "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                          "focusable": "false",
                          "role": "presentation",
                          "viewBox": "0 0 24 24",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "d": "M7 10l5 5 5-5z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "role": undefined,
                          "viewBox": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "role": undefined,
                          "viewBox": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-haspopup": undefined,
                      "aria-pressed": undefined,
                      "class": undefined,
                      "role": undefined,
                      "tabindex": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-haspopup": undefined,
                      "aria-pressed": undefined,
                      "class": undefined,
                      "role": undefined,
                      "tabindex": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "id": "station-select",
                      "name": "stationID",
                      "type": "hidden",
                      "value": "",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                        "focusable": "false",
                        "role": "presentation",
                        "viewBox": "0 0 24 24",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "d": "M7 10l5 5 5-5z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "role": undefined,
                        "viewBox": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "role": undefined,
                        "viewBox": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "aria-haspopup": "true",
                        "aria-pressed": "false",
                        "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                        "role": "button",
                        "tabindex": "0",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {},
                          "children": Array [
                            Object {
                              "data": "â€‹",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-haspopup": undefined,
                        "aria-pressed": undefined,
                        "class": undefined,
                        "role": undefined,
                        "tabindex": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-haspopup": undefined,
                        "aria-pressed": undefined,
                        "class": undefined,
                        "role": undefined,
                        "tabindex": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "id": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "id": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "MuiSvgIcon-root-57 MuiSelect-icon-26",
                      "focusable": "false",
                      "role": "presentation",
                      "viewBox": "0 0 24 24",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "d": "M7 10l5 5 5-5z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "id": "station-select",
                        "name": "stationID",
                        "type": "hidden",
                        "value": "",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "aria-haspopup": "true",
                          "aria-pressed": "false",
                          "class": "MuiSelect-select-21 MuiSelect-selectMenu-24 MuiInputBase-input-50 MuiInput-input-35",
                          "role": "button",
                          "tabindex": "0",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "data": "â€‹",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-haspopup": undefined,
                          "aria-pressed": undefined,
                          "class": undefined,
                          "role": undefined,
                          "tabindex": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-haspopup": undefined,
                          "aria-pressed": undefined,
                          "class": undefined,
                          "role": undefined,
                          "tabindex": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "id": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "focusable": undefined,
                      "role": undefined,
                      "viewBox": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "focusable": undefined,
                      "role": undefined,
                      "viewBox": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "MuiFormLabel-root-13 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10",
                "data-shrink": "false",
                "for": "station-select",
              },
              "children": Array [
                Object {
                  "data": "Station",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-shrink": undefined,
                "for": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-shrink": undefined,
                "for": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
